sympy/core/relational.py

```diff
@@ -302,9 +302,15 @@ def equals(self, other, failing_expression=False):
 
     def _eval_simplify(self, **kwargs):
         from .add import Add
+        from sympy.core.expr import Expr
         r = self
         r = r.func(*[i.simplify(**kwargs) for i in r.args])
         if r.is_Relational:
+            if not isinstance(r.lhs, Expr) or not isinstance(r.rhs, Expr):
+                result = _eval_is_eq(r.lhs, r.rhs)
```

Comment:
This should use `is_eq` rather than `_eval_is_eq` but actually this shouldn't need to be here at all. The `r.func(...)` call above should be checking `is_eq`.

I think all we need to do here is avoid doing anything unless both sides are Expr.